modal-js/src/queue.ts

```diff
@@ -0,0 +1,464 @@
+  ): Promise<Queue_> {
+    var request = client.queueGetOrCreate({
+      deploymentName: name,
+      objectCreationType: createIfMissing
+        ? ObjectCreationType.OBJECT_CREATION_TYPE_CREATE_IF_MISSING
+        : undefined,
+      namespace: DeploymentNamespace.DEPLOYMENT_NAMESPACE_WORKSPACE,
+      environmentName: environmentName(environment_name),
+    });
+    var resp = await request;
+    return new Queue_(resp.queueId);
+  }
+
+  static async delete(
+    name: string,
+    { environment_name }: { environment_name?: string } = {},
+  ): Promise<void> {
+    var obj = await Queue_.fromName(name, { environment_name });
+    await retryTransientErrors({
+      fn: client.queueDelete,
+      args: [{ queueId: obj.objectId }],
+    });
+  }
+
+  async _get_nonblocking({
+    partition,
+    n_values,
+  }: {
+    partition?: string;
+    n_values: number;
+  }): Promise<any[]> {
+    var request = {
+      queueId: this.objectId,
+      partitionKey: Queue_.validatePartitionKey(partition),
+      timeout: 0,
+      nValues: n_values,
+    };
+
+    var response = await retryTransientErrors({
+      fn: client.queueGet,
+      args: [request],
+    });
+    if (response.values) {
+      return response.values.map((value) => loads(value));
+    } else {
+      return [];
+    }
+  }
+
+  async _get_blocking({
+    partition,
+    timeout,
+    n_values,
+  }: {
+    partition?: string;
+    timeout?: number;
+    n_values: number;
+  }): Promise<any[]> {
+    var deadline: number | undefined = undefined;
+    if (timeout !== undefined) {
+      deadline = timeNow() + timeout;
+    }
+
+    while (true) {
+      var requestTimeout = 50.0;
+      if (deadline) {
+        requestTimeout = Math.min(requestTimeout, deadline - timeNow());
+      }
+
+      var request = {
+        queueId: this.objectId,
+        partitionKey: Queue_.validatePartitionKey(partition),
+        timeout: requestTimeout,
+        nValues: n_values,
+      };
+
+      var response = await retryTransientErrors({
+        fn: client.queueGet,
+        args: [request],
+      });
+
+      if (response.values && response.values.length > 0) {
+        return response.values.map((value) => loads(value));
+      }
+
+      if (deadline && timeNow() > deadline) {
+        break;
+      }
+    }
+
+    throw new EmptyError("Queue is empty");
+  }
+
+  async clear({
+    partition,
+    all,
+  }: {
+    partition?: string;
+    all?: boolean;
+  }): Promise<void> {
+    if (partition && all) {
+      throw new InvalidError(
+        "Partition must be null when requesting to clear all.",
+      );
+    }
+
+    var request = {
+      queueId: this.objectId,
+      partitionKey: Queue_.validatePartitionKey(partition),
+      allPartitions: all,
+    };
+
+    await retryTransientErrors({
+      fn: client.queueClear,
+      args: [request],
+    });
+  }
+
+  async get({
+    block = true,
+    timeout,
+    partition,
+  }: {
+    block?: boolean;
+    timeout?: number;
```

Comment:
Question for @freider : why would we have `block` option that can be set to false, if block defaults to true, and setting block to false is equivalent to setting `timeout` to zero? Is this redundant?